server/topic.go

```diff
@@ -223,9 +225,14 @@ func (t *Topic) userIsReader(uid types.Uid) bool {
 
 // maybeFixTopicName sets the topic field in `msg` depending on the uid.
 func (t *Topic) maybeFixTopicName(msg *ServerComMessage, uid types.Uid) {
-	if !uid.IsZero() && t.cat == types.TopicCatP2P {
+	// For zero uids we don't know the proper topic name.
+	if uid.IsZero() {
+		return
+	}
+
+	if t.cat == types.TopicCatP2P || (t.cat == types.TopicCatGrp && t.isChan) {
```

Comment:
For channel topics, should we also check what the session's original subscription was?
Say, if the session subscribed to 'grpXYZ', we'll be returning 'chnXYZ' here.

Reply:
I would need to pass another argument to the method for a very minor efficiency improvement. It just does not seem to be worth it. The `t.original(uid)` will produce the correct topic name without knowing the original session's topic name.